game-core/src/test/java/games/strategy/triplea/attachments/TriggerAttachmentTest.java

```diff
@@ -26,102 +29,226 @@
 import games.strategy.engine.data.PlayerId;
 import games.strategy.engine.data.ProductionFrontier;
 import games.strategy.engine.data.ProductionFrontierList;
 import games.strategy.engine.data.ProductionRule;
 import games.strategy.engine.data.ProductionRuleList;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.TestAttachment;
 import games.strategy.engine.delegate.IDelegateBridge;
 import games.strategy.engine.history.IDelegateHistoryWriter;
 
 @ExtendWith(MockitoExtension.class)
 class TriggerAttachmentTest {
 
-  @Mock
-  private IDelegateBridge bridge;
-  @Mock
-  private IDelegateHistoryWriter historyWriter;
-
-  @BeforeEach
-  void setUp() {
-    final GameData gameData = new GameData();
-
-    when(bridge.getData()).thenReturn(gameData);
-    when(bridge.getHistoryWriter()).thenReturn(historyWriter);
+  @Nested
+  class TriggerChangeTest {
+
+    @Mock
+    private IDelegateBridge bridge;
+    @Mock
+    private IDelegateHistoryWriter historyWriter;
+
+    @BeforeEach
+    void setUp() {
+      final GameData gameData = new GameData();
+
+      when(bridge.getData()).thenReturn(gameData);
+      when(bridge.getHistoryWriter()).thenReturn(historyWriter);
+    }
+
+    @Test
+    void testTriggerProductionFrontierEditChange() {
+      final GameData gameData = bridge.getData();
+      final TriggerAttachment triggerAttachment = mock(TriggerAttachment.class);
+      final Set<TriggerAttachment> satisfiedTriggers = Collections.singleton(triggerAttachment);
+
+      when(triggerAttachment.getProductionRule())
+          .thenReturn(Arrays.asList("frontier:rule1", "frontier:-rule2", "frontier:rule3"));
+      when(triggerAttachment.getName()).thenReturn("mockedTriggerAttachment");
+
+      final ProductionRuleList productionRuleList = gameData.getProductionRuleList();
+      productionRuleList.addProductionRule(new ProductionRule("rule1", gameData));
+      final ProductionRule productionRule2 = new ProductionRule("rule2", gameData);
+      productionRuleList.addProductionRule(productionRule2);
+      productionRuleList.addProductionRule(new ProductionRule("rule3", gameData));
+
+      final ProductionFrontierList productionFrontierList = gameData.getProductionFrontierList();
+      productionFrontierList
+          .addProductionFrontier(
+              new ProductionFrontier("frontier", gameData, Collections.singletonList(productionRule2)));
+
+      TriggerAttachment.triggerProductionFrontierEditChange(
+          satisfiedTriggers,
+          bridge,
+          "beforeOrAfter",
+          "stepName",
+          false, // useUses
+          false, // testUses
+          false, // testChance
+          false); // testWhen
+      final ArgumentCaptor<CompositeChange> argument = ArgumentCaptor.forClass(CompositeChange.class);
+      verify(bridge).addChange(argument.capture());
+      final CompositeChange change = argument.getValue();
+      assertFalse(change.isEmpty());
```

Comment:
Interesting, I thought we decided to not use `ArgumentCaptor` in this particular case (not talking about the case below, that one seems reasonable to me) in the last PR.
There `not(argThat(Change::isEmpty))` was used as argument matcher. Was there a particular problem to this approach?

Reply:
I apologize, I forgot to change it in the existing test, it should have been fixed now.